### PR TITLE
TF controller update

### PR DESCRIPTION
**Pull request by k0da**

### CODE REVIEW

File: pkg/executor/terraform/terraform.go [hunk 1]
```diff
@@ -53,12 +70,16 @@ func Plan(destroy bool) (string, error) {
...
+	output, err := shell.Execute(cmd)
 	if err != nil {
 		return "", err
 	}
 
-	return combineOutput(output), nil
+	return output, nil
 }
 
 func combineOutput(in []string) string {
```
Comment (somaritane) [inline pkg/executor/terraform/terraform.go]: Looks like `combineOutput()` is not being used anywhere anymore, should it be deleted?

Comment (k0da) [reply pkg/executor/terraform/terraform.go]: Yeah.

Comment (k0da) [reply pkg/executor/terraform/terraform.go]: done

**Review by ytsarev — CHANGES_REQUESTED** [review]
Please consider using the following linting approach https://github.com/AbsaOSS/k8gb/blob/master/Makefile#L148-L152

```
make lint
staticcheck ./...
-: could not analyze dependency github.com/rancher/terraform-controller/pkg/executor/runner of github.com/rancher/terraform-controller/pkg/executor (compile)
-: could not analyze dependency github.com/rancher/terraform-controller/pkg/executor/terraform of github.com/rancher/terraform-controller/pkg/executor/runner (compile)
pkg/executor/terraform/terraform.go:5:2: "strings" imported but not used (compile)
make: *** [lint] Error 1
```

Code definitely requires some cleanup - currently it even fails to compile with unused module errror

**Comment by k0da** [comment]
Yes, a code need a little bit more love. Added Makefile target... Fixed code compilation.

**Comment by ytsarev** [comment]
```
make lint
staticcheck ./...
pkg/executor/main.go:76:2: this value of out is never used (SA4006)
make: *** [lint] Error 1
```
as of now, but I guess you are on it

**Review by kuritka — CHANGES_REQUESTED** [review]
Hi, can we push without `/vendor` folder?
 
Screenshot 2020-10-14 at 10 16 22

All necessary references lives in `go.mod`. Commands `go mod download` or `go mod tidy` installs modules into local mod cache and don't need to be stored on GitHub .

**Comment by k0da** [comment]
@kuritka I understand your concern. it is quite hard to review during updates/upgrades. But at the same time IMO this is protects us from an internet connection being broken or upstream for module is gone for some reason.

**Comment by ytsarev** [comment]
@k0da in the early days of k8gb we also used `/vendor` and deprecated eventually deprecated this approach due to lot of inconveniences during upgrades and mess in the PRs . I strongly suggest to switch to clean `go mod`

**Comment by kuritka** [comment]
@k0da good point here. If internet connectivity is the reason, then we should move to modules.  After `go mod download` the files being cached in ` $GOPATH/pkg/mod/cache` and being there until `go mod clean`. Compiler looks into that offline folder. 
So from the local development compiling, it is equal. From the self-runner it is true. If data-center lost connection to the internet, the Go cannot create `$GOPATH/pkg/mod/cache` and compilation fails. But I think we can rely on the data-center connectivity. 

Here are some reasons for `/vendor`

- keep references next to the code, so when reference package dismiss, you still have access to package in the form you need. Someone has to solve this complication one day, anyway.
- keep builds cheap, In case we build in the cloud and pay for traffic (the vendor folder may take hundreds of MB and downloading each time increase costs).  Building cache is a time-consuming operation we should consider if we pay for run-time.
-  `GO1.11`,`GO1.12` supported  modules preliminary so to be safe `/vendor` might be an option. But from the PR I see we switched to `GO1.13` (maybe we can [switch to](https://blog.golang.org/go1.15) `GO1.15` if everything goes well).



The problem I have with vendor except repo robustness are nightmare PR's and conflicts when bumping package versions. Also risk that new package download and IDE will change it by mistaken refactoring.

I think we should use vendor in case we deal with the point I mentioned ahead (i.e. we pay for the traffic).

We can consider [Go module proxy](https://arslan.io/2019/08/02/why-you-should-use-a-go-module-proxy/) (artifactory :) ) which may help in many scenarios.

**Comment by somaritane** [comment]
> We can consider [Go module proxy](https://arslan.io/2019/08/02/why-you-should-use-a-go-module-proxy/) which may help in many scenarios.
@k0da , @kuritka , afaik go uses https://proxy.golang.org by default, starting from 1.13

**Comment by k0da** [comment]
vendor dropped, staticcheck fixed, I'll leave errcheck errors for later.

**Review by ytsarev — APPROVED** [review]

**Review by kuritka — CHANGES_REQUESTED** [review]
Hi, maybe you oversight something and  didn't correctly drop vendor. 

Screenshot 2020-10-14 at 23 14 20

**Pull request merged**